### DISCUSSION TITLE
Make Lambda functions testable, allow dependencies, and get rid of CFN inline 4096 chars limit

### DISCUSSION THREAD

**Discussion by s0enke** [superwerker/superwerker#192] [Ideas]
We had a call AWS Quick Starts team and discussed how they handle regional assets, e.g. Lambda functions. 

In addition to #189, we discovered two more possible ways:

1. The Quick Start approach is to copy regional assets as a CFN custom resource before rolling out the Lambda functions -> https://github.com/aws-quickstart/lambda-copyzips/blob/main/templates/copy-zips.template.yaml
2. We also discussed if we could use CDK. The idea is that the CFN template bootstraps e.g. a CodeBuild which then in turn does cdk deploy . This is also OK for the Quick Starts team.
3. Use Systems Manager Automation documents with`executeScript` instead of Lambdas (this would solve the 4096 char problem, but not Make Lambda functions testable, allow dependencies).

I think (1) is faster to implement for the short-term (e.g. to solve the 4096 char problem in the dashboard(#189)). (2) for the mid/long term (e.g. 2.0 release)

**Comment by LukvonStrom** [comment]
Regarding approach 2: I would suggest delivering a [cloud assembly](https://docs.aws.amazon.com/cdk/latest/guide/apps.html#apps_cloud_assembly) as an artefact (e.g. the files when one performs `cdk synth` ) and deploy them via pipeline only. This would result in less overhead and a clean build compared to a pipeline locally synthing and deploying (with additional dependency installs etc.)

**Comment by s0enke** [comment]
CDK integration is done in #280

**Reply by LukvonStrom** [reply]
@s0enke very nice! Over the time I came to appreciate the simplicity of cdk-pipeline, which would remove the custom ressource (if I'm not mistaken) you're currently using in your implementation. Switching is not too difficult.
